sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -3491,11 +3490,19 @@ pi_result piKernelCreate(pi_program Program, const char *KernelName,
   ze_kernel_handle_t ZeKernel;
   ze_result_t ZeResult = ZE_RESULT_ERROR_INVALID_KERNEL_NAME;
   _pi_program::ModuleIterator ModIt(Program);
+  uint32_t KernelNum = 0;
```

Comment:
This variable is only used inside the `while` loop, so move it's declaration there.